force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,16 +122,11 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
+        Integer numVowels;
+        numVowels = (str.toLowerCase()).countMatches('a')+(str.toLowerCase()).countMatches('e')+(str.toLowerCase()).countMatches('i')
+        + (str.toLowerCase()).countMatches('o') + (str.toLowerCase()).countMatches('u');
 
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
-        }
-        */
-
-        return null; // Replace null with the variable you used to store the result
+        return numVowels; 
```

Comment:
This is the most unique solution I have seen to this problem. I forgot there was a countMatches method on the String class.  But I have a feeling it is not very performant. I don't know how countMatches I implemented, but my guess is that it is checking every member of the string every time.  So, let's say you get a 10-letter string. You will check every character 8 times for 80 iterations.  See my solution below with a Set. 

```
  public static Integer countVowels(String str) {
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        List<String> characters = str.split('');
        for (String character : characters) {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
 ```